### PR TITLE
adding if condition to remove keys

### PR DESCRIPTION
**Pull request by vuillaut**
DL1 to DL2 fails on merged DL1 files where images have been removed.

Fixed this and added unit tests for merged DL1 files.

### CODE REVIEW

**Comment by MaxNoe** [comment]
I would suggest a bug fix release, since no other PRs have been merged yet.

File: lstchain/scripts/tests/test_lstchain_scripts.py [hunk 1]
```diff
@@ -103,6 +104,27 @@ def test_lstchain_mc_rfperformance():
     assert os.path.exists(file_model_energy)
 
 
+@pytest.mark.run(after='test_lstchain_mc_r0_to_dl1')
+def test_lstchain_merge_dl1_hdf5_files():
+    run_program('lstchain_merge_hdf5_files',
```
Comment (MaxNoe) [inline lstchain/scripts/tests/test_lstchain_scripts.py:109]: Where does it get it's input files from?

**Comment by MaxNoe** [comment]
Apperently the next version of sklearn has been released with the removal of joblib. You just need to replace the imports of joblib with `import joblib` and add it to the dependencies.

File: lstchain/scripts/tests/test_lstchain_scripts.py (hunk 1 above)
Comment (vuillaut) [reply lstchain/scripts/tests/test_lstchain_scripts.py:109]: merge all hdf5 files in the dir passed with `-d`.
here `output dir`

Comment (MaxNoe) [reply lstchain/scripts/tests/test_lstchain_scripts.py:109]: Ah, was confused by the `output_dir`, so here it's actually the input

**Comment by vuillaut** [comment]
> Apperently the next version of sklearn has been released with the removal of joblib. You just need to replace the imports of joblib with `import joblib` and add it to the dependencies.

shall we fix the version of joblib to ensure compatibility of the models?

**Comment by MaxNoe** [comment]
@vuillaut I don't think joblib affects compatibility. I tested this for the aict tools and there were no problems with loading models stored with older versions of joblib with newer versions of joblib.

The story is different for sklearn, so you should always load a model with the same version of sklearn that was used to train it.

**Comment by rlopezcoto** [comment]
> Apperently the next version of sklearn has been released with the removal of joblib. You just need to replace the imports of joblib with import joblib and add it to the dependencies.

The `joblib` issue is addressed here:
https://github.com/cta-observatory/cta-lstchain/pull/338/files
as soon as [this](https://github.com/cta-observatory/cta-lstchain/pull/338/files#r425006523) is addressed, it can be merged.

**Comment by MaxNoe** [comment]
As the other PR also does other changes, I would rather merge this here fast, that be simpler, right?

**Comment by rlopezcoto** [comment]
> As the other PR also does other changes, I would rather merge this here fast, that be simpler, right?

but without doing any joblib modification, right? Then I agree (as soon as it passes travis checks)

**Comment by MaxNoe** [comment]
@vuillaut with pickle protocol 5 (python 3.8) and numpy 1.18, joblib becomes obsolete and we can just use the stdlib pickle module

**Comment by MaxNoe** [comment]
> but without doing any joblib modification, right? Then I agree (as soon as it passes travis checks)

Why? no need to revert this. The changes are even exactly the same, so it won't even create a merge conflict.

**Comment by morcuended** [comment]
> damn, that one was missed... it means that for this production, we either have to merge at a dl2 level or release a bugfix version for the processing... @morcuended were you thinking on performing the merge before producing the `DL2`?

No, we also produce DL2 files subrun-wise

**Comment by morcuended** [comment]
Is the problem with missing joblib module fixed in this PR?

**Comment by rlopezcoto** [comment]
> Why? no need to revert this. The changes are even exactly the same, so it won't even create a merge conflict.

I did not see that it was already [included](https://github.com/cta-observatory/cta-lstchain/pull/409/commits/e35c3d65ead280d89cf8770adc22fa959cf8cce9). So fine then

**Comment by vuillaut** [comment]
- Concerning joblib, I made the changes already don't worry
- Concerning merge, it's a problem for the IRF production at least (and needs to be fixed anyway)

**Comment by rlopezcoto** [comment]
> Is the problem with missing joblib module fixed in this PR?

should be

> No, we also produce DL2 files subrun-wise

it seems that @vuillaut needs this, so I'll do the v0.5.1 release and install it in the IT cluster

**Comment by vuillaut** [comment]
> @vuillaut with pickle protocol 5 (python 3.8) and numpy 1.18, joblib becomes obsolete and we can just use the stdlib pickle module

good to know.
To keep in mind when we support only python >= 3.8 ;-)

**Comment by vuillaut** [comment]
Ready for review.

**Comment by vuillaut** [comment]
And we are celebrating 100th unit test 🥳

**Pull request merged**